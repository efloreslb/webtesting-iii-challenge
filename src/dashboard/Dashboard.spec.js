import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/react/cleanup-after-each';
import 'jest-dom/extend-expect';

import Dashboard from './Dashboard';

describe('<Dashboard />', () => {

   it('renders without crashing', () => {
      render(<Dashboard />);
   })

   it('matches snapshot', () => {
      const snap = renderer.create(<Dashboard />);
      expect(snap.toJSON()).toMatchSnapshot();
   })

   it('defaults to unlocked and open, options are lock and close', () => {
      const { getByText } = render(<Dashboard />);
      const open = getByText(/open/i);
      const unlocked = getByText(/unlocked/i);

      expect(open).toBeInTheDocument();
      expect(unlocked).toBeInTheDocument();
   })

   it('buttons and display text changes to reflect the state the door will be in if clicked', () => {
      const { getByTestId } = render(<Dashboard />);
      
      const toggleclosed = getByTestId('closeButton');
      const displayclosed = getByTestId('closedOpen');

      fireEvent.click(toggleclosed);

      expect(displayclosed).toHaveTextContent('Closed');
      expect(toggleclosed).toHaveTextContent('Open Gate');

      const togglelocked = getByTestId('lockButton');
      const displaylocked = getByTestId('lockedUnlocked');

      fireEvent.click(togglelocked);

      expect(displaylocked).toHaveTextContent('Locked');
      expect(togglelocked).toHaveTextContent('Unlock Gate');
   })


})