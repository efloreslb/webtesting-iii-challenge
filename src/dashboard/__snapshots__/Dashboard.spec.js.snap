// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="lockedUnlocked"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="closedOpen"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockButton"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="closeButton"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
